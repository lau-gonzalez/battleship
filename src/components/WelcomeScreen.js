import React from 'react'

export default function WelcomeScreen(props) {
    return (
        <div className='container-fluid d-flex bg-primary welcome justify-content-center flex-row align-items-center'>
            <div className="col-md-4 text-center">
                <h1 className='mb-4'>Battleship</h1>
                <form onSubmit={e => {e.preventDefault(); props.onSubmitWelcome(e.target.welcome.value); props.history.push('/options') }}>
                    <input type="text" name='welcome' className='form-control mb-3' placeholder='Enter Name' required/>
                    <button type='submit' className='btn btn-dark'>
                        Start Battle!
                    </button>
                </form>
            </div>            
        </div>
    )
}
