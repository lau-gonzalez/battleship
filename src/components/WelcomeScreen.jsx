/* eslint-disable react/prop-types */
/* eslint-disable max-len */
/* eslint linebreak-style: ["error", "windows"] */
import React from 'react';
import PropTypes from 'prop-types';
import ship from '../img/ship.png';

export default function WelcomeScreen(props) {
  const { onSubmitWelcome } = props;
  return (
    <div className="container-fluid d-flex bg-primary welcome justify-content-center flex-column align-items-center">
      <img src={ship} alt="ship" />
      <div className="col-md-4 text-center">
        <h1 className="mb-4">Battleship</h1>
        <form onSubmit={(e) => { e.preventDefault(); onSubmitWelcome(e.target.welcome.value); props.history.push('/battleship'); }}>
          <input type="text" name="welcome" className="form-control mb-3" placeholder="Enter Name" required />
          <button type="submit" className="btn btn-dark">
            Start Battle!
          </button>
        </form>
      </div>
    </div>
  );
}

WelcomeScreen.propTypes = {
  onSubmitWelcome: PropTypes.func.isRequired,
};
