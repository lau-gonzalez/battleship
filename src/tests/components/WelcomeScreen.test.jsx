/* eslint-disable no-console */
/* eslint-disable no-undef */
/* eslint linebreak-style: ["error", "windows"] */
import React from 'react';
import { shallow } from 'enzyme';
import WelcomeScreen from '../../components/WelcomeScreen';

describe('../../components/WelcomeScreen', () => {
  it('should work', () => {
    const onSubmitWelcome = (input) => console.log(input);
    const wrapper = shallow(
      <WelcomeScreen
        onSubmitWelcome={onSubmitWelcome}
      />,
    );
    console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });
});
