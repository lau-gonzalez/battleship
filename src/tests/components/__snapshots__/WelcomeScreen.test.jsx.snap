// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../../components/WelcomeScreen should work 1`] = `ShallowWrapper {}`;
